commit 98e702da92ed0f0f5acf048091d247dec2d19d29 (refs/changes/63/2963/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-11-17T12:34:07+01:00 (1 year, 11 months ago)
    
    TOOLS-1858 Want sdcadm updates channel to set based on installer

diff --git a/answers.json.tmpl b/answers.json.tmpl
index 7656e1af..b264a920 100644
--- a/answers.json.tmpl
+++ b/answers.json.tmpl
@@ -27,5 +27,6 @@
     "api_password": "joypass123",
     "mail_to": "<default>",
     "mail_from": "<default>",
-    "phonehome_automatic": "<default>"
+    "phonehome_automatic": "<default>",
+    "update_channel": "release"
 }
diff --git a/answers.json.tmpl.external b/answers.json.tmpl.external
index 38f495c1..aac46362 100644
--- a/answers.json.tmpl.external
+++ b/answers.json.tmpl.external
@@ -33,5 +33,6 @@
     "api_password": "joypass123",
     "mail_to": "<default>",
     "mail_from": "<default>",
-    "phonehome_automatic": "<default>"
+    "phonehome_automatic": "<default>",
+    "update_channel": "release"
 }
diff --git a/scripts/prompt-config.sh b/scripts/prompt-config.sh
index 6751830a..ec4c66c8 100755
--- a/scripts/prompt-config.sh
+++ b/scripts/prompt-config.sh
@@ -1199,6 +1199,32 @@ Joyent to help us make SmartDataCenter better.
 	phonehome_automatic="$val"
 
 
+	printheader "Update Channel"
+	message="
+Name of the update channel to be used for this Datacenter.
+One of 'support', 'release' or 'dev'.
+Customers with a support contract must select the 'support' channel.
+\n"
+
+	if [[ $(getanswer "skip_instructions") != "true" ]]; then
+		printf "$message"
+	fi
+
+	while [ true ]; do
+		key="update_channel"
+		promptval "Enter the update channel" \
+			"$update_channel" "${key}"
+		if [[ "$val" == "dev" || "$val" == "release" || \
+					"$val" == "support" ]]; then
+			update_channel="$val"
+			break
+		fi
+		echo "Must select one of 'support', 'release' or 'dev'"
+		# disable key, since this means bad value in answer file
+		key=
+	done
+
+
 	printheader "Verify Configuration"
 	message=""
 
@@ -1244,6 +1270,8 @@ Joyent to help us make SmartDataCenter better.
 		echo
 		printf "Enable telemetry: $phonehome_automatic\n"
 		echo
+		printf "Update channel: $update_channel\n"
+		echo
 	fi
 
 	if [[ $(getanswer "skip_final_confirm") != "true" ]]; then
@@ -1660,6 +1688,8 @@ echo >>$tmp_config
 
 echo "phonehome_automatic=${phonehome_automatic}" >>$tmp_config
 
+echo "update_channel=${update_channel}" >>$tmp_config
+
 # Always show the timers and make setup serial for now.
 echo "show_setup_timers=true" >> $tmp_config
 if [[ $(getanswer "dtrace_zone_setup") == "true" ]]; then
